SharedSources/VLCRemoteControlService.m

```diff
@@ -18,17 +18,21 @@ @implementation VLCRemoteControlService
 static inline NSArray * RemoteCommandCenterCommandsToHandle()
 {
     MPRemoteCommandCenter *cc = [MPRemoteCommandCenter sharedCommandCenter];
-    return @[cc.pauseCommand,
-             cc.playCommand,
-             cc.stopCommand,
-             cc.togglePlayPauseCommand,
-             cc.nextTrackCommand,
-             cc.previousTrackCommand,
-             cc.skipForwardCommand,
-             cc.skipBackwardCommand,
-             cc.changePlaybackRateCommand,
-             cc.changePlaybackPositionCommand,
-             ];
+    NSMutableArray *commands = [NSMutableArray arrayWithObjects:
+                                cc.pauseCommand,
+                                cc.playCommand,
+                                cc.stopCommand,
+                                cc.togglePlayPauseCommand,
+                                cc.nextTrackCommand,
+                                cc.previousTrackCommand,
+                                cc.skipForwardCommand,
+                                cc.skipBackwardCommand,
+                                cc.changePlaybackRateCommand,
+                                nil];
+    if (@available(iOS 9.1, *)) {
+        [commands addObject:cc.changePlaybackPositionCommand];
+    }
+    return commands;
```

Comment:
Wouldn't that mean the receiver has to typecast the `NSArray` to a `NSMutableArray` for this to be a problem?

Reply:
I've checked and Apple guidelines say:

> To determine whether it can change a received object, the receiver of a message must rely on the formal type of the return value. If it receives, for example, an array object typed as immutable, it should not attempt to mutate it.

[Source](https://developer.apple.com/library/content/documentation/General/Conceptual/CocoaEncyclopedia/ObjectMutability/ObjectMutability.html#//apple_ref/doc/uid/TP40010810-CH5-SW67)

Reply:
yes they would have to typecast it and yes they shouldn't attempt to mutate it. Adding a copy is still good practice in this case since it prevents them from being able to do that. You will also find this practice in Apple's code examples :)